src/mobilenet.jl

```diff
@@ -0,0 +1,304 @@
+
+backbone(m::MobileNetv2) = m.layers[1]
+classifier(m::MobileNetv2) = m.layers[2:end]
+
+# MobileNetv3
+
+"""
+Hard-Swish activation function
+([reference](https://arxiv.org/abs/1905.02244)).
+"""
+@inline h_swish(x) = x * hardσ(x)
+
+"""
+Squeeze and Excitation layer used by MobileNetv3
+([reference](https://arxiv.org/abs/1905.02244)).
+"""
+function selayer(channels, reduction = 4)
+  SkipConnection(
+    Chain(
+        AdaptiveMeanPool((1, 1)),
+        conv_bn((1, 1), channels, channels // reduction, relu; bias = false)...,
+        conv_bn((1, 1), channels // reduction, channels, hardσ)...,
+    ), .*
+  )
+end
+
+"""
+    invertedresidualv3(inplanes, hidden_planes, outplanes, kernel_size, stride, use_se, use_hs)
+
+Create a basic inverted residual block for MobileNetv3
+([reference](https://arxiv.org/abs/1905.02244)).
+
+# Arguments
+- `inplanes`: number of input feature maps
+- `hidden_planes`: number of feature maps in the hidden layer
+- `outplanes`: number of output feature maps
+- `kernel_size`: kernel size of the convolutional layers
+- `stride`: stride of the convolutional kernel, has to be either 1 or 2
+- `use_se`: if `true`, Squeeze and Excitation layer will be used
+- `use_hs`: if `true`, Hard-Swish activation function will be used
+"""
+function invertedresidualv3(inplanes, hidden_planes, outplanes, kernel_size,
+                            stride, use_se, use_hs)
+  @assert stride in [1, 2] "`stride` has to be 1 or 2"
+
+  if inplanes == hidden_planes
+    invres = Chain(
+      conv_bn((kernel_size, kernel_size), hidden_planes, hidden_planes, use_hs ? h_swish : relu;
+               bias = false, stride, pad = (kernel_size - 1) ÷ 2, groups = hidden_planes)...,
+      use_se ? selayer(hidden_planes) : identity,
+      conv_bn((1, 1), hidden_planes, outplanes, identity; bias = false)...
+    )
+  else
+    invres = Chain(
+      conv_bn((1, 1), inplanes, hidden_planes, use_hs ? h_swish : relu; bias = false)...,
+      conv_bn((kernel_size, kernel_size), hidden_planes, hidden_planes, use_hs ? h_swish : relu;
+               bias = false, stride, pad = (kernel_size - 1) ÷ 2, groups = hidden_planes)...,
+      use_se ? selayer(hidden_planes) : identity,
+      conv_bn((1, 1), hidden_planes, outplanes, identity; bias = false)...
+    )
+  end
+
+  (stride == 1 && inplanes == outplanes) ? SkipConnection(invres, +) : invres
+end
+
+"""
+    mobilenetv3(configs, mode, width_mult; nclasses = 1000)
+
+Create a MobileNetv3 model.
+([reference](https://arxiv.org/abs/1905.02244)).
+
+# Arguments
+- `configs`: a list of configurations for each layer that details 
+- `mode`: 'large' or 'small'
+- `width_mult`: Controls the number of channels in each layer, with 1.0 being the original
+  model as detailed in the paper.
+- `nclasses`: the number of output classes
+"""
+function mobilenetv3(configs, mode, width_mult; nclasses = 1000)
+  @assert mode in ["large", "small"] "`mode` has to be either \"large\" or \"small\""
```

Comment:
Prefer to use `Symbol`s instead of `String`s.

Reply:
Have done. I'm a little new to the concept of metaprogramming though, so what exactly is the difference here? And why are `Symbols` preferred?

Reply:
I don't really think there's a big difference here, but Julia conventions typically use `Symbol`s for arguments of this form. `Symbol`s do have a special meaning in metaprogramming, but in this context, they are effectively just another data type (i.e. we are not doing any metaprogramming).